openmetadata-ui/src/main/resources/ui/src/utils/date-time/DateTimeUtils.ts

```diff
@@ -236,39 +236,53 @@ export const calculateInterval = (
   }
 };
 
-const intervals: [string, number][] = [
-  ['Y', 933120000000], // 1000 * 60 * 60 * 24 * 30 * 360
-  ['M', 2592000000], // 1000 * 60 * 60 * 24 * 30
-  ['d', 86400000], // 1000 * 60 * 60 * 24
-  ['h', 3600000], // 1000 * 60 * 60
-  ['m', 60000], // 1000 * 60
-  ['s', 1000], // 1000
-];
-
 /**
  * Converts a given time in milliseconds to a human-readable format.
  *
  * @param milliseconds - The time duration in milliseconds to be converted.
  * @returns A human-readable string representation of the time duration.
  */
 export const convertMillisecondsToHumanReadableFormat = (
-  milliseconds: number,
+  ms: number,
   length?: number
 ): string => {
-  if (milliseconds <= 0) {
+  if (ms <= 0) {
     return '0s';
   }
 
+  const duration = Duration.fromMillis(ms);
   const result: string[] = [];
-  let remainingMilliseconds = milliseconds;
-
-  for (const [name, count] of intervals) {
-    if (remainingMilliseconds < count) {
-      continue; // Skip smaller units
-    }
-    const value = Math.floor(remainingMilliseconds / count);
-    remainingMilliseconds %= count;
-    result.push(`${value}${name}`);
+
+  // Extract each unit from the duration
+  const years = Math.floor(duration.as('years'));
+  const months = Math.floor(duration.as('months')) % 12;
+  const days = Math.floor(duration.as('days')) % 30;
+  const hours = Math.floor(duration.as('hours')) % 24;
+  const minutes = Math.floor(duration.as('minutes')) % 60;
+  const seconds = Math.floor(duration.as('seconds')) % 60;
+  const milliseconds = Math.floor(duration.as('milliseconds')) % 1000;
+
+  // Add non-zero units to the result
+  if (years > 0) {
+    result.push(`${years}Y`);
+  }
+  if (months > 0) {
+    result.push(`${months}M`);
+  }
```

Comment:
please double check this, its being used in DQ side please valide the behaviour for the same

Reply:
The only behaviour change is we have also included the `milliseconds` display in the string. I have checked it's effects on DQ, and nothing looks concerning. But I'll confirm again. Thanks